src/ansys/fluent/core/solver/flobject.py

```diff
@@ -215,9 +215,31 @@ def max(self):
 class Textual(Property):
     """Exposes attribute accessor on settings object - specific to string objects."""
 
+    _items_with_string_and_allowed_values = set()
+
+    def _filter_items_with_string_and_allowed_values(self, data):
+        for key, value in data.items():
+            if isinstance(value, dict):
+                if (
+                    "type" in value
+                    and "has_allowed_values" in value
+                    and value["type"] == "string"
+                    and value["has_allowed_values"] is True
+                ):
+                    self._items_with_string_and_allowed_values.add(key)
+                self._filter_items_with_string_and_allowed_values(value)
+
     def allowed_values(self):
         """Get the allowed values of the object."""
-        return self.get_attr("allowed-values", (list, str))
+
+        static_info = self.flproxy.get_static_info()
+        self._filter_items_with_string_and_allowed_values(static_info)
+
+        if self.obj_name in self._items_with_string_and_allowed_values:
+            try:
+                return self.get_attr("allowed-values", (list, str))
+            except BaseException as ex:
+                return []
```

Comment:
We check that if obj_name is in `items_with_string_and_allowed_values` then only we return allowed_values.

Reply:
I thought this would be implemented via mixin approach.

Reply:
Done. Thank you.